### DISCUSSION TITLE
How to enable CORS?

### DISCUSSION THREAD

**Discussion by francelwebdev** [mojolicious/mojo.js#95] [Q&A] [answered]
How to enable cors ?

**Comment by jberger** [comment]
There are several [plugins on CPAN](https://metacpan.org/search?size=20&q=Mojolicious+cors) or you can set the [headers yourself](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)

**Reply by kraih** [reply]
This is the mojo.js section. 😉

**Reply by jberger** [reply]
Oops, I need to be more careful to check that!

**Comment by kraih** [comment, marked as answer]
If you hadn't left IRC so abruptly i would have given you a more detailed explanation than you got until then. [CORS](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) is just a bunch of headers you could set like `ctx.res.set('Access-Control-Allow-Origin', '*');`. Ideally you would implement CORS as router [conditions](https://mojojs.org/docs/Routing.md#adding-conditions) in an NPM plugin. We have [a bunch](https://metacpan.org/pod/Mojolicious::Plugin::SecureCORS) for Perl already, porting one of them should be very easy.

To enable CORS indiscriminately for all routes you could just add this hook to your application:
```js
app.addContextHook('router:before', async ctx => {
 ctx.res.set('Access-Control-Allow-Origin', '*');
});
```

**Reply by francelwebdev** [reply]
Hello Sir, ok thanks, please add an option to simply enable cors, a bit like in express with: app.use(cors()).

mojo({
   // Enable cors
   cors: true
});

**Comment by francelwebdev** [comment]
Hi, how to set cors with multiple domain ?
Also, how to generate open api swagger doc for an api ?

Thanks
